sumpy/expansion/multipole.py

```diff
@@ -86,10 +86,13 @@ def coefficients_from_source(self, kernel, avec, bvec, rscale, sac=None):
                 rscale, (1,), sac=sac)
 
     def evaluate(self, kernel, coeffs, bvec, rscale, sac=None):
+        from sumpy.tools import DifferentiatedExprDerivativeTaker
         if not self.use_rscale:
             rscale = 1
 
         base_taker = kernel.get_derivative_taker(bvec, rscale, sac)
+        base_taker = DifferentiatedExprDerivativeTaker(base_taker,
+                {tuple([0]*self.dim): 1})
```

Comment:
Why is this helpful? (IIUC this should be equivalent to calling the underlying derivative taker?) If not, add a comment to explain.

Reply:
It's for easier code to maintain. If you look at https://github.com/inducer/sumpy/blob/97fdea77d71a7e32e0c8eaa7b7beb885ea9e1ef4/sumpy/kernel.py#L1035 there are two code branches.
1. DifferentiatedExprDerivativeTaker instances
2. Regular sympy expressions

If I didn't have this here, we would have a third code branch,

3. DerivativeTaker instances that are not DifferentiatedExprDerivativeTaker instances

Reply:
Thx!